lib/util.mli

```diff
@@ -24,6 +24,15 @@ val on_fst : ('a -> 'b) -> 'a * 'c -> 'b * 'c
 val on_snd : ('a -> 'b) -> 'c * 'a -> 'c * 'b
 val map_pair : ('a -> 'b) -> 'a * 'a -> 'b * 'b
 
+(** Folding under pairs *)
+
+val fold_fst : ('c -> 'a -> 'c * 'a) -> 'c -> 'a * 'b -> 'c * ('a * 'b)
+val fold_snd : ('c -> 'b -> 'c * 'b) -> 'c -> 'a * 'b -> 'c * ('a * 'b)
+
+(** Equality on pairs *)
+
+val pair_equal : ('a -> 'a -> bool) -> ('b -> 'b -> bool) -> ('a * 'b -> 'a * 'b -> bool)
```

Comment:
This is also called `eq_pair` in micromega mltutils

Reply:
Thanks for the remark. I renamed it to `eq_pair` (but let micromega decide on its own if it wants to use the new copy in util.ml).